library/common-library/templates/_naming.tpl

```diff
@@ -8,6 +8,27 @@ This is an function to be called directly with a string just to truncate strings
 
 
 
+{{- /*
+Given a name and a suffix returns a 'DNS Valid' which always include the suffix, truncating the name if needed.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+It fails if suffix is too long to fint in the limit set.
+Usage:
+{{ include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "<my-name>" "suffix" "my-suffix" ) }}
+*/ -}}
+{{- define "newrelic.common.naming.truncateToDNSWithSuffix" -}}
+{{- $maxLen := (sub 63 (len .suffix)) -}}
+
+{{- if (lt $maxLen 1) -}}
+    {{ fail "(newrelic.common.naming.truncateToDNSWithSuffix) suffix length is too long to compose a valid name" }}
```

Comment:
NIT: This error is not specific enough and is going to be shown to customers... It should be more specific.
```suggestion
    {{ fail (printf "The release name you choose (%s) is hitting the Kubernetes limit of 63 chars for the object name. We cannot create %s-%s" .name .name .suffix) }}
```
I did not test my suggestion but you should get the idea ;)

Reply:
The problem would not be the release name but the suffix chosen instead, anyway I like the idea of being more specific, thanks!

Reply:
Error improved here: [b4ecebf](https://github.com/newrelic/helm-charts/pull/785/commits/b4ecebf7ba33d0cdefd6f0d8708d1d4413248759)